process/tech-due-diligence-guidelines.md

```diff
@@ -0,0 +1,103 @@
+* make sure you've read, in detail, the relevant [project proposal](https://github.com/quinton-hoole/toc/tree/master/proposals).
+  This will usually be in the form of an [open pull request](https://github.com/cncf/toc/pulls).
+  I'd suggest holding off on commenting on the PR until you've completed the next three steps.
+* take a look at some [previous submissions](https://github.com/cncf/toc/pulls?utf8=%E2%9C%93&q=is%3Apr)
+  (both successful and unsuccessful) to help calibrate your expectations.
+* do as much reading up as you need to (and consult with experts in the specific field) in order to familiarize yourself with the technology
+  landscape in the immediate vicinity of the project (and don't only use the proposal and that project's documentation as a guide in this regard).
+* at this point you should have a very clear technical idea of what exactly the project actually does and does not do, roughly how it compares with and differs from 
+  similar projects in it's technology area, and/or a set of unanswered questions in those regards. 
+* go through the [graduation criteria](https://www.cncf.io/projects/graduation-criteria/) and for each item,
+  decide for yourself whether or not you have enough info to make a strong, informed call on that item.
+  * If so, write it down, with motivation.
+  * If not, jot down what information you feel you're missing.
+  * Also take note of what unanswered questions the community might have posted in the PR review that you consider
+    to be critically important.
+    
+### Some example questions that will ideally need clear answers
+
+#### Technical
+
+* What are the current performance and scalability bounds of the software?  Have these been explicitly tested?
+* What exactly are the failure modes?  Are they well understood?  Have they been tested?  Do they form part of continuous integration testing?
+* What trade-offs have been made regarding performance, scalability, complexity, reliability, security etc?  Are these trade-offs explicit or implicit?  Why?
+* Code quality.  Does it look good, bad or mediocre to you (based on a spot review).  How thorough are the code reviews? Substance over form.
+  Are there explicit coding guidelines for the project?
+* What is the release model?  Versioning scheme?  Evidence of stability or otherwise of past stable released versions?
+* What is the CI/CD status?  Do explicit code coverage metrics exist? If not, what is the subjective adequacy of automated testing?  Do different levels of tests exist (e.g. unit, integration, interface, end-to-end), or is there only partial coverage in this regard?  Why? 
+
+#### Team
+
+* What is the general quality of informal communication around the project (slack, github issues, PR reviews, technical blog posts, etc)?
+* Would you hire the team? (this is of course a purely hypothetical question).  If not, why?
+* How much time do they commit to the project?
+* How big is the team? Who funds them? Why? How much? For how long?
```

Comment:
Please tie this back to criteria. Presumably this is to address whether the project has "a healthy number of committers"

Reply:
and "community size and any existing sponsorship"

Reply:
I added a link to the criteria.  I'm not sure that it's valuable to tie every question back to a specific criterion. Or are you suggesting that there are questions here that are irrelevant?  Many of the questions came from @monadic so he might want to comment on this.